zeppelin/converter.py

```diff
@@ -0,0 +1,212 @@
+        types, add the file type to the dictionary and create the necessary
+        function to handle it.
+
+        """
+        output_options = {
+            'HTML': self.build_image,
+            'TEXT': self.build_text,
+            'TABLE': self.build_table
+        }
+
+        if 'editorMode' in paragraph['config']:
+            mode = paragraph['config']['editorMode'].split('/')[-1]
+            if paragraph['results']['msg']:
+                msg = paragraph['results']['msg'][0]
+                if mode not in ('text', 'markdown'):
+                    output_options[msg['type']](msg)
+
+    def parse_date(self, date):
+        return datetime.strptime(date, '%b %d, %Y %I:%M:%S %p')
+
+    def process_date_created(self, text):
+        """ Sets date_created to the oldest date """
+        if self.date_created == 'N/A':
+            self.date_created = text
+        if self.parse_date(text) < self.parse_date(self.date_created):
+            self.date_created = text
+
+    def process_date_updated(self, text):
+        """ Sets date_updated to the most recent date """
+        if self.date_updated == 'N/A':
+            self.date_updated = text
+        if self.parse_date(text) > self.parse_date(self.date_updated):
+            self.date_updated = text
+
+    def process_title(self, text):
+        self.out.append('#### ' + text)
+
+    def build_markdown_body(self, text):
+
+        """ Generate the body for the Markdown file
+
+        - processes each json block one by one
+        - for each block
+            - process the input by detecting the editor language
+            - print the input
+            - process the output by detecting the output format
+            - print the output
+
+        """
+        for paragraph in text['paragraphs']:
+            if 'user' in paragraph:
+                self.user = paragraph['user']
+            if 'dateCreated' in paragraph:
+                self.process_date_created(paragraph['dateCreated'])
+            if 'dateUpdated' in paragraph:
+                self.process_date_updated(paragraph['dateUpdated'])
+            if 'title' in paragraph:
+                self.process_title(paragraph['title'])
+            if 'text' in paragraph:
+                self.process_input(paragraph['text'])
+            if 'results' in paragraph:
+                self.process_results(paragraph)
+
+    def build_output(self, fout):
+        fout.write('\n'.join([s.encode('utf-8') for s in self.out]))
+
+    def convert(self):
+
+        """
+        Takes in a .json file as input and convert it to Markdown format,
+        saving the generated .png images into ./images.
+
+        """
+        try:
+            with open(self.input_filename, 'rb') as raw:
+                t = json.load(raw)
+                full_path = os.path.join(self.directory, self.output_filename + self.MD_EXT)
+                with open(full_path, 'w') as fout:
+                    self.build_markdown_body(t)   # create the body
```

Comment:
Please put the comments either on the line above, or using the following format:
```python
self.build_markdown_body(t)  # create the body
```